pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-hadoop/src/main/java/org/apache/pinot/plugin/ingestion/batch/hadoop/HadoopSegmentGenerationJobRunner.java

```diff
@@ -230,7 +230,10 @@ public void run()
       if (hadoopTokenFileLocation != null) {
         jobConf.set("mapreduce.job.credentials.binary", hadoopTokenFileLocation);
       }
-      jobConf.setInt(JobContext.NUM_MAPS, numDataFiles);
+
+      int jobParallelism =
+          (_spec.getSegmentCreationJobParallelism() > 0) ? _spec.getSegmentCreationJobParallelism() : numDataFiles;
```

Comment:
Should we use max of configured parallelism and the `numDataFiles`?

Reply:
You are right :)